compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/JvmCreateTypeGen.java

```diff
@@ -324,7 +324,8 @@ private MethodVisitor createPopulateTypeMethod(ClassWriter cw, String methodName
     public void addImmutableType(MethodVisitor mv, BType type) {
         BIntersectionType immutableType = ((SelectivelyImmutableReferenceType) type).getImmutableType();
         if (immutableType == null || type.tsymbol == null || immutableType.tsymbol == null ||
-                !(immutableType.tsymbol.pkgID.equals(type.tsymbol.pkgID))) {
+                !(immutableType.tsymbol.pkgID.equals(type.tsymbol.pkgID))
```

Comment:
Had to filter it at the type gen; because if we filter out while setting the immutable type at the ImmutableTypeCloner; multiple calls to `setImmutableType` for the same type would add multiple type definitions. If we are to avoid adding multiple type definitions; we'll have to either iterate through the type defs and check for the original type or maintain the previously created immutable types. Hence; went with this approach

Reply:
Will this work when the module source doesn't have the intersection type?

e.g.,

Source has a record type `Foo` but no intersection `Foo & readonly`. But the tests have `Foo & readonly`?

Reply:
Yes. It was the actal issue. I've added tests under testerina tests

Reply:
In the immutable type cloner we check for package ID atm, right? Wouldn't checking package ID + is testable there, work? I don't get why we have to go through type definitions?

Btw, do we have a test for when `Foo` and `Foo & readonly` are in the tests also? IIRC, we did.

Reply:
Added a test. Even though we do ID+testable check, we'll need to filter the test pkg references from the main pkg.